### PR TITLE
Improvements in keeping or destroying models

### PR DESCRIPTION
**Pull request by addyess**
introduce Enumerations to preserve or destroy various tracked models

### CODE REVIEW

File: pytest_operator/plugin.py [hunk 1]
```diff
@@ -566,10 +594,8 @@ def cloud_name(self) -> Optional[str]:
     @property
     def keep_model(self) -> bool:
         """Represents whether the current model should be kept after tests."""
-        if self._init_keep_model:
-            return True
         current_state = self.current_alias and self._models.get(self.current_alias)
-        return current_state.keep if current_state else False
+        return current_state.keep if current_state else self._init_keep_model
```
Comment (addyess) [inline pytest_operator/plugin.py:598]: So long as the model exists, `track_model` will ensure the value of `keep` is set according to the spec.

If the current_alias is unset or the model isn't found -- fall back to the global flag.

File: pytest_operator/plugin.py [hunk 2]
```diff
@@ -730,7 +756,7 @@ async def track_model(
         model_name: Optional[str] = None,
         cloud_name: Optional[str] = None,
         use_existing: Optional[bool] = None,
-        keep: Optional[bool] = None,
+        keep: Union[ModelKeep, str, bool, None] = ModelKeep.IF_EXISTS,
```
Comment (addyess) [inline pytest_operator/plugin.py:759]: So many options 😄

File: pytest_operator/plugin.py [hunk 3]
```diff
@@ -768,19 +795,35 @@ async def track_model(
         if model_name and use_existing is None:
             use_existing = await self._model_exists(model_name)
 
-        keep = bool(use_existing) if keep is None else keep
+        keep_val: bool = False
+        if isinstance(keep, str):
+            keep = OpsTest.ModelKeep(keep.lower())
```
Comment (addyess) [inline pytest_operator/plugin.py:800]: map strings into enum values

File: pytest_operator/plugin.py [hunk 4]
```diff
@@ -746,11 +772,12 @@ async def track_model(
...
-               None:  follows the value of use_existing
-               False: tracked model is destroyed at tests end
-               True:  tracked model remains once tests complete
-               --keep-models flag will override this options
+        @param Optional[ModelKeep, str, bool, None] keep:
+               ModelKeep  : See docs for the enum
+               str        : mapped to ModelKeep values
+               None       : Same as ModelKeep.IF_EXISTS
+               True       : Same as ModelKeep.ALWAYS
+               False      : Same as ModelKeep.NEVER, but respects keep-models flag
```
Comment (mateoflorido) [inline pytest_operator/plugin.py:780]: I think the behavior has changed, right? 
```suggestion
               False      : Same as ModelKeep.NEVER, overrides the keep-models flag
```

Comment (addyess) [reply pytest_operator/plugin.py:780]: The previous `False` setting here would destroy models at the end of the test -- unless `keep-models` flag was set.  I have to maintain that `False` will repeat that behavior

Comment (addyess) [reply pytest_operator/plugin.py:780]: if you want to override the `keep-models` flag.  just use `ops_test.ModelKeep.NEVER`

**Review by mateoflorido — APPROVED** [review]
LGTM!

**Pull request merged**